### PR TITLE
[WIP]Cyborg Riding and Spinning

### PR DESCRIPTION
**Pull request by silicons**
TODO: 
FIND A WAY TO LET PEOPLE PULL YOU OFF THE CYBORG

You can now ride on a cyborg by buckling yourself to it. Cyborgs can also buckle people to itself.
Cyborgs now have the _spin emote that lets them spin around quickly, likely throwing off anyone riding on it.
You can not buckle simple bots to it like Beepsky due to balance reasons, immershuns reasons, meme reasons, and most of all, BUGGED REASONS.
You can not buckle handcuffed mobs to it as that would be pretty overpowered.
:cl:
rscadd: Cyborgs have been equipped with handgrips. Why Nanotrasen did this, no one will ever know. You can now buckle yourself to a cyborg to ride them.
rscadd: Cyborgs now have programming to spin in place. This is likely defence to assistants using them as sentient skateboards. (_spin to most likely throw off anyone riding you as a cyborg)
rscdel: The addition of handgrips has made Cyborgs too bulky to be buckled in
rscdel: A recent firmware update has rendered more simple robots like Securitrons, Cleanbots, and others unable to be buckled to anything. 
/:cl:

### CODE REVIEW

**Comment by Zilenan91** [comment]
I can't even imagine how disgusting the movable buckle cuffs would be with secborgs

**Comment by kevinz000** [comment]
securitrons is because movement with them buckled to you lets them move you and not the other way around, and bucklecuffing is going to lead to cyborgcurity with beepskies and shit.

**Comment by xxalpha** [comment]
Can the cyborg push the rider off?

**Comment by kevinz000** [comment]
that's what spin is for :^)

**Comment by kevinz000** [comment]
W H E W B A C K TO  BUGFIX

**Comment by AdamElTablawy** [comment]
>drag beepsky
>bucklecuff the person you arrested

### jesus christ how horrifying

**Comment by InsaneHyena** [comment]
> Shitsec riding a janiborg. Officer spills the blood with harmbaton, janiborg cleans it up.
This is stuff of nightmares.

**Comment by RemieRichards** [comment]
You're not offsetting the rider's pixel offsets at all, which will result in the 3(3!?!?!) people riding it to stack on top of each other like weirdos.

**Comment by kevinz000** [comment]
remie, one person on each side, but not the face. also how do i do this "pixel offsetting"

**Comment by RemieRichards** [comment]
Now you've made it buckleable, you can define some buckle procs on it.
Like how scooters push you up 4 pixels, look up buckling in the code.

**Comment by Supermichael777** [comment]
I'M TAKING THE CYBORG, JOIN ME IN GLORY!

File: code/modules/mob/living/silicon/silicon.dm [hunk 1]
```diff
@@ -9,6 +9,9 @@
 	verb_yell = "alarms"
 	see_in_dark = 8
 	bubble_icon = "machine"
+	can_buckle = 1
+	max_buckled_mobs = 3
+	buckle_lying = 0
```
Comment (phil235) [inline code/modules/mob/living/silicon/silicon.dm]: buckling people to AI?

Comment (Shadowlight213) [inline code/modules/mob/living/silicon/silicon.dm]: You should totally be able to buckle the AI to a borg tbh. Borgs can already drag AIs around, having one ride the borg would be hilarious.

Comment (RemieRichards) [inline code/modules/mob/living/silicon/silicon.dm]: @Shadowlight213 re-read.

Comment (Shadowlight213) [inline code/modules/mob/living/silicon/silicon.dm]: Ah.

**Comment by PKPenguin321** [comment]
@kevinz000 Look at skateboard code, it has a lot of what you want, including pixel shifts and throwing people off.

**Comment by Iamgoofball** [comment]
ridin spinners

they dont stop

**Comment by drynwyn** [comment]
are merge tokens transferable

**Comment by coiax** [comment]
@kevinz000 I might suggest making spin an action we grant cyborgs, rather than hiding an important feature (ahelp HELP HOW DO I GET [insert appropriate assistant name] OFF OF ME) under an emote

**Comment by drynwyn** [comment]
Or just include a message when someone mounts you that says you can use spin.

**Comment by Cheridan** [comment]
This sounds like a really cute and funny idea in theory that is probably going to be abused a shitload in practice.

**Comment by PKPenguin321** [comment]
make it so if you buckle a stunned/sleeping human it falls off since it can't hold onto the handlebars

**Comment by PKPenguin321** [comment]
Hey kevinz if you want a merge token try fixing https://github.com/tgstation/tgstation/issues/13209 supposedly it's really easy and it should buff your favorite sec-related roles

**Comment by kevinz000** [comment]
tfw merge conflicts

**Comment by optimumtact** [comment]
If you finish your pr faster, I can merge it faster there will be less conflicts faster

**Comment by optimumtact** [comment]
F to pay respects

**Comment by kevinz000** [comment]
It's not dead im spending time on autismcraft :^)

**Comment by Armhulen** [comment]
no, trust me you are

**Comment by optimumtact** [comment]
reopen or ask for reopen when ready

**Pull request closed without merging**